sspl_test/functional_tests/debug.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2001-2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""
+ ****************************************************************************
+  Description:       Flags and methods for handling debug mode
+ ****************************************************************************
+"""
+
+import json
+from framework.utils.service_logging import logger
+
+try:
+    from systemd import journal
+
+    use_journal = True
+except ImportError:
+    use_journal = False
+
+
+class Debug(object):
+    """ "Flags and methods for handling debug mode"""
+
+    def __init__(self):
+        super(Debug, self).__init__()
+        self._debug = False
+        self._debug_persist = False
+
+    def _log_debug(self, message):
+        """Logging messages"""
+        if self._debug:
+            log_msg = self.name() + ", " + message
+            if use_journal:
+                journal.send(log_msg, PRIORITY=7, SYSLOG_IDENTIFIER="sspl-ll")
+            else:
+                logger.info(log_msg)
+
+    def _set_debug(self, debug):
+        """Sets debug flag"""
+        if not self._debug_persist:
+            self._debug = debug
+
+    def _get_debug(self):
+        """Returns the debug flag"""
+        return self._debug
+
+    def _set_debug_persist(self, debug_persist):
+        """Sets debug persist flag"""
+        self._debug_persist = debug_persist
+
+    def _get_debug_persist(self):
+        """Returns debug persist flag"""
+        return self._debug_persist
+
+    def _disable_debug_if_persist_false(self):
+        """Turn debug mode off if persistence is False"""
+        if not self._debug_persist:
+            self._debug = False
+
+    def _check_debug(self, jsonMsgRaw):

```

Comment:
Does this exist in current test framework. I remember seeing some SSPL code enabling debug mode based on debug flags in json data, though how its helping with SSPL now? If not getting used in current test framework, lets avoid this. Basically not clear how this will help